Assets/Scripts/Ability/Combat/SmiteAbility.cs

```diff
@@ -0,0 +1,35 @@
+using RogueApeStudio.Crusader.Units;
+using UnityEngine;
+using RogueApeStudio.Crusader.HealthSystem;
+using System;
+using System.Collections.Generic;
+
+namespace RogueApeStudio.Crusader.Ability.Combat
+{
+    [CreateAssetMenu(fileName = "SmiteAbility", menuName = "Ability/Line Area Ability/Smite Ability")]
+    internal class SmiteAbility : LineAreaAbility
+    {
+        public override void Cast(Vector3 position, Vector3 direction)
+        {
+            Collider[] hitColliders = CreateArea(position, Quaternion.LookRotation(direction));
+           
+            foreach (var hitCollider in hitColliders)
+            {
+                if (hitCollider.CompareTag("Enemy"))
+                {
+                    Enemy enemyComponent = hitCollider.GetComponent<Enemy>();
+                    EnemyHealth healthComponent = hitCollider.GetComponent<EnemyHealth>();
+
+                    if (enemyComponent == null)
+                    {
+                        Debug.LogWarning("Enemy component is null, skipping.");
```

Comment:
Why is it not throwing an `ArgumentNullException`?

Reply:
full if statement is gone